### ISSUE TITLE
cloudprovider_azure metrics not available

### ISSUE THREAD

**Issue by koceg** [kubernetes-sigs/azurefile-csi-driver#453]
Image:`mcr.microsoft.com/k8s/csi/azurefile-csi:v0.9.0`

After following the example at https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/7d0c79c328807ba1d5a350652f5538d0027f67f6/deploy/example/metrics no metric of cloudprovider_azure was available.

Before trying this example  I provisioned a pvc claim using dynamic provisioning and assigned it to a pod that was running during this test.
```
PVC
Name:          azure-pvc
Namespace:     demo1
StorageClass:  azurefile-csi
Status:        Bound
Volume:        pvc-8bfb4559-XXXX-XXXX-XXXX-9c3757e925e9
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: file.csi.azure.com
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      10Gi
Access Modes:  RWX
VolumeMode:    Filesystem
Mounted By:    nginx-pv2
```

I connected to one of the pods using cmd `oc port-forward csi-azurefile-controller-xxxx-xxxx 29614`
and when I connected to url `http://localhost:29614/metrics` I didn't see any metrics that contained cloud,provider or azure in their name.

is there something else that needs to be done first?

**Comment by andyzhangx** [comment]
it works by this way:
```
# kubectl exec -it csi-azurefile-controller-568756fbf8-rkh5c -n kube-system -c azurefile sh
# curl http://localhost:29614/metrics
# HELP go_gc_duration_seconds A summary of the GC invocation durations.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 8.701e-06
go_gc_duration_seconds{quantile="0.25"} 1.9402e-05
go_gc_duration_seconds{quantile="0.5"} 6.0304e-05
go_gc_duration_seconds{quantile="0.75"} 9.7307e-05
go_gc_duration_seconds{quantile="1"} 0.002913717
...
```

create a service with `ClusterIP` type should also work: https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/deploy/example/metrics

**Comment by andyzhangx** [comment]
should not be an issue

**Comment by koceg** [comment]
I tried your example but I don't know what kubectl command should demonstrate?

As there is no curl command in the image (assuming that the second command should be executed from within the pod itself)
that being said I am able to connect to the service and I'm able to retrieve the metrics.

But the metrics from file `vendor/k8s.io/legacy-cloud-providers/azure/metrics/azure_metrics.go` don't seem to be exported.(assumption might be wrong)

As a comparison image `quay.io/mavazque/reversewords:latest` does similar thing and exposes metrics on `IP:8080/metrics` with a specific custom metric `total_reversed_words` that I'm able to work with. I expected the same behavior from image`mcr.microsoft.com/k8s/csi/azurefile-csi:v0.9.0` when I query the metrics.

**Comment by ZeroMagic** [comment]
@koceg The comment in another issue may be helpful. https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/505#issuecomment-679858771

**Comment by andyzhangx** [comment]
One major issue is that you may have two controllers(one is leader, another is secondary), in that case, you could only get metris from leader controller. I have checked metrics from file `vendor/k8s.io/legacy-cloud-providers/azure/metrics/azure_metrics.go`, the metrics is there if you curl from leader controller.

```console
# curl http://$ip:29604/metrics | grep api_request_duration_seconds
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP cloudprovider_azure_api_request_duration_seconds [ALPHA] Latency of an Azure API call
# TYPE cloudprovider_azure_api_request_duration_seconds histogram
cloudprovider_azure_api_request_duration_seconds_bucket{request="disks_create_or_update",resource_group="mc_andy-aks1190_andy-aks1190_eastus2euap",source="",subscription_id="b9d2281e-dcd5-4dfd-9a97-0d50377cdf76",le="0.1"} 0
cloudprovider_azure_api_request_duration_seconds_bucket{request="disks_create_or_update",resource_group="mc_andy-aks1190_andy-aks1190_eastus2euap",source="",subscription_id="b9d2281e-dcd5-4dfd-9a97-0d50377cdf76",le="0.25"} 0
cloudprovider_azure_api_request_duration_seconds_bucket{request="disks_create_or_update",resource_group="mc_andy-aks1190_andy-aks1190_eastus2euap",source="",subscription_id="b9d2281e-dcd5-4dfd-9a97-0d50377cdf76",le="0.5"} 0
cloudprovider_azure_api_request_duration_seconds_bucket{request="di
```

**Comment by koceg** [comment]
True. Have tested and yes , once I query the leader I get the metrics.
Thank you for clarification
